src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+
+            // Read back the values from the configuration register which should now be default.
+            regValue = ReadRegister(Register.Configuration, 1000);
+
+            PowerLsb = PowerLsb = 1F;
+            _calibrationValue = 0;
+
+            // Get the current values for the Bus and Shunt ADC settings.
+            _busAdcResSamp = (AdcResolutionOrSamples)((regValue & (ushort)ConfigurationFlags.BadcMask) >> 4);
+            _shuntAdcResSamp = (AdcResolutionOrSamples)(regValue & (ushort)ConfigurationFlags.SadcMask);
+        }
+
+        /// <summary>
+        /// Set the Ina219 operating mode.
+        /// </summary>
+        /// <param name="mode">The operating mode to switch to.</param>
+        public void SetOperatingMode(OperatingMode mode)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.ModeMask ^ 0xFFFF;
+            regValue |= (ushort)mode;
+
+            WriteRegister(Register.Configuration, regValue);
+        }
+
+        /// <summary>
+        /// Set the Ina219 Bus voltage range.
+        /// </summary>
+        /// <param name="range">The voltage range to switch to.</param>
+        public void SetBusVoltageRange(BusVoltageRange range)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.BrngMask ^ 0xFFFF;
+            regValue |= (ushort)range;
+
+            WriteRegister(Register.Configuration, regValue);
+        }
+
+        /// <summary>
+        /// Set the Ina219 Programable Gain Amplifier range.
+        /// </summary>
+        /// <remarks>
+        /// This is the voltage range used to read the shunt voltage. It can be one of +/-40mV, +/-80mV, +/-160mV or +/-320mV.
+        /// </remarks>
+        /// <param name="sensitivity">The voltage range of the Programable Gain Amplifier to switch to.</param>
+        public void SetPGARange(PgaSensitivity sensitivity)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.PgaMask ^ 0xFFFF;
+            regValue |= (ushort)sensitivity;
+
+            WriteRegister(Register.Configuration, regValue);
+        }
+
+        /// <summary>
+        /// Set the Ina219 ADC resolution or samples to be used when reading the Bus voltage.
+        /// </summary>
+        /// <remarks>
+        /// This can either by the number of bits used for the ADC conversion (9-12 bits) or the number of samples at 12 bits to be averaged for the result.
+        /// </remarks>
+        /// <param name="resorsamp">The ADC resolution or sample count to switch to.</param>
+        public void SetBusAdcResolutionOrSamples(AdcResolutionOrSamples resorsamp)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.BadcMask ^ 0xFFFF;
```

Comment:
should you ve `&` the mask instead of flipping bits?

Reply:
I’m anding with the inverse of the mask